scipy/linalg/basic.py

```diff
@@ -85,11 +143,13 @@ def solve(a, b, sym_pos=False, lower=False, overwrite_a=False,
         Whether to check that the input matrices contain only finite numbers.
         Disabling may give a performance gain, but may result in problems
         (crashes, non-termination) if the inputs do contain infinities or NaNs.
-    assume_a : str, optional
-        Valid entries are explained above.
-    transposed: bool, optional
-        If True, ``a^T x = b`` for real matrices, raises `NotImplementedError`
-        for complex matrices (only for True).
+    sym_pos : bool, optional
```

Comment:
Yes, deprecating this sounds good.

Reply:
We need @pv 's blessing for that as he didn't like the idea of dropping it previously.

Reply:
Would it be easier to create a new function?
I know that `solve` it the obvious name, and it's a shame that it would not be available. But maybe create a new function (`backslash`?) for now, deprecate the things we don't like about `solve`, and after a few versions make `solve` and the new function synonymous?
Maybe it's a crazy idea, but I've been tiptoeing around some unfortunate parts of the interface of `svds` recently and thinking how nice it would be to just start fresh.

Reply:
I know exactly how you feel and I feel the same about many linalg parts but I think these are pretty central to usage of SciPy. That is to say, there are too many users importing scipy just for linalg and I don't know how to handle them. 

Every time I sit down for some work on this, I get lost in a backwards compatible API design mess and I rage quit which happens every weekend which is the only actual coding time I can spare recently.

If I find a way to regularize the input then, say, I can't handle transposed keyword. If I do that then Cython doesn't like certain inputs, if we fix that then we lose too much time checking too many things then it becomes pointless to implement the whole thing etc. So I think a few more weekends and there will be some progress on this.

Reply:
> these are pretty central to usage of SciPy. That is to say, there are too many users importing scipy just for linalg and I don't know how to handle them.

Sure. So maybe forget the part about deprecating the bad parts of `solve`; just consider the part about making this a new function. If the name were `backslash`, it would go at the very top of the `linalg` documentation:
![image](https://user-images.githubusercontent.com/6570539/126083296-d9ac842b-1b73-41d6-957f-f6f8520504cb.png)

After all, this function is going to do more than strictly "solve" a linear system, since it would also handle the non-square case, right?

Reply:
Is it not too counterintuitive to call this `backslash`? Wouldn't something like `ldivide` be clearer? (That might actually be the MATLAB name).

Sorry, I can only contribute to the colour of the bikeshed :)

Reply:
It might be for non-matlab users. I didn't think other people would agree with it, but I thought it would be fun : )